### PR TITLE
Add CORS support to sda-auth #minor

### PR DESCRIPTION
**Pull request by norling**
This PR adds Cross-Origin Resourse Sharing support to the sda-auth service, as well as a CORS testing page to the dev-server.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,19 +20,46 @@ Parameter | Description | Defined value
...
+`CORS_CREDENTIALS` | If cookies, authorization headers, and TLS client certificates are allowed over CORS | `false`
 `SERVER_CERT` | Certificate file path | `""`
 `SERVER_KEY` | Private key file path | `""`
 `S3INBOX` | S3 inbox host | `s3.example.com`
 
 ## Running the development setup
 
 First, create a RSA private key under the name `sign-rsa-jwt.key` in the folder `keys`.
+```
+openssl genrsa -out keys/sign-rsa-jwt.key 2048
```
Comment (aaperis) [inline README.md:34]:
```suggestion
openssl genrsa -out keys/sign-rsa-jwt.key 2048  && chmod 664 keys/sign-rsa-jwt.key
```

**Review by aaperis — CHANGES_REQUESTED** [review]
I tried the minimal login test site but the Login gave `error: redirect_uri_mismatch
`. Could it be something missing from the configuration?

**Comment by norling** [comment]
Good catch! :) I had indeed missed documenting that you need to update the `CLIENT_REDIRECT_URI` to use the CORS login.

**Review by aaperis — APPROVED** [review]
Looks good :-)

File: README.md [hunk 2]
```diff
@@ -20,19 +20,49 @@ Parameter | Description | Defined value
...
+
+The current setup also requires that
+```
+127.0.0.1  oidc
 ```
+is added to `/etc/hosts`, so that routing works properly.
+
+## Running with Cross-Origin Resource Sharing (CORS)
+
+This servce can be run as a backend only, and in the case where the frontend
```
Comment (dbampalikis) [inline README.md:51]:
```suggestion
This service can be run as a backend only, and in the case where the frontend
```

File: README.md [hunk 3]
```diff
@@ -20,19 +20,49 @@ Parameter | Description | Defined value
...
+Recommended cors settings for a given host are:
+```
+export CORS_ORIGINS="https://<frontend-url>"
+export CORS_METHODS="GET,OPTIONS"
+export CORS_CREDENTIALS="true"
+```
+
+There is a minimal CORS login testing site at http://localhost:8000 of the
+dev-server. To use the CORS login page, the `dev-server/docker-compose.yml` file
+must be updated by settings `CLIENT_REDIRECT_URI` (under
```
Comment (dbampalikis) [inline README.md:63]:
```suggestion
must be updated by setting the `CLIENT_REDIRECT_URI` (under
```

File: README.md [hunk 4]
```diff
@@ -20,19 +20,49 @@ Parameter | Description | Defined value
...
+`services/oidc/environment`) to `http://localhost:8000` and the server
+restarted.
```
Comment (dbampalikis) [inline README.md:65]:
```suggestion
`services/oidc/environment`) to `http://localhost:8000` and restart the server.
```

File: main.go [hunk 5]
```diff
@@ -182,12 +188,20 @@ func (auth AuthHandler) getEGAConf(ctx iris.Context) {
...
+	redirectURI := ctx.Request().URL.Query().Get("redirect_uri")
+	if redirectURI != "" {
+		redirectParam := oauth2.SetAuthURLParam("redirect_uri", redirectURI)
+		ctx.Redirect(auth.OAuth2Config.AuthCodeURL(state.String(), redirectParam))
+	} else {
+		ctx.Redirect(auth.OAuth2Config.AuthCodeURL(state.String()))
+	}
 }
 
 // getElixirLogin authenticates the user with return values from the elixir
```
Comment (dbampalikis) [inline main.go:201]:
```suggestion
// elixirLogin authenticates the user with return values from the elixir
```

**Review by dbampalikis — APPROVED** [review]
Apart for some really minor comments, it looks good!

**Pull request merged**